packages/input/src/Input/Input.tsx

```diff
@@ -0,0 +1,157 @@
+import { Slot as RadixSlot } from "@radix-ui/react-slot";
+import { useComposedRefs } from "@telegraph/compose-refs";
+import clsx from "clsx";
+import React from "react";
+
+import { COLOR, SIZE } from "./Input.constants";
+
+type BaseRootProps = {
+  size?: keyof typeof SIZE.Container;
+  variant?: keyof typeof COLOR.Container.default;
+  error?: boolean;
+};
+
+type RootProps = BaseRootProps &
+  Omit<React.ComponentPropsWithoutRef<"input">, "size">;
+
+type RootRef = HTMLInputElement;
+
+type InternalProps = Omit<BaseRootProps, "error"> & {
+  state: "default" | "disabled" | "error";
+};
+
+type Required<T> = {
+  [P in keyof T]-?: T[P];
+};
+
+const InputContext = React.createContext<Required<InternalProps>>({
+  state: "default",
+  size: "2",
+  variant: "outline",
+});
+
+const Root = React.forwardRef<RootRef, RootProps>(
+  (
+    {
+      size = "2",
+      variant = "outline",
+      className,
+      disabled,
+      error,
+      children,
+      ...props
+    },
+    forwardedRef,
+  ) => {
+    const inputRef = React.useRef<HTMLInputElement>(null);
+    const composedRefs = useComposedRefs(forwardedRef, inputRef);
+
+    const state = disabled ? "disabled" : error ? "error" : "default";
```

Comment:
Derives state for displaying the right visuals on the input. Disabled is then passed into the input to get out of the box browser functionality.